### PR TITLE
Addon-actions: safe render for cyclic obj

### PR DESCRIPTION
**Pull request by Armanio**
Issue: #6004

## What I did
Wrap deepEqual function call in try/catch.
I spend much time finding an elegant solution, but...
I analyzed comparing objects and they have cyclic react instance ('__reactInternalInstance...' has an element, an element has __reactblabla, and etc ). I don't know its bug or feature.

More information: 'fast-deep-equal' issue and author reply [here](https://github.com/epoberezkin/fast-deep-equal/issues/8)

And I rewrite 'addAction' method: if new state based on the previous state you must use a callback in 'setState' function. React [docs](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) for more information.

## How to test
Start cra-kitchen-app
Open button sample
Double click to button

- Is this testable with Jest or Chromatic screenshots?
no
- Does this need a new example in the kitchen sink apps?
no
- Does this need an update to the documentation?
no

### CODE REVIEW

**Review by shilman — APPROVED** [review]
LGTM? Will wait for another opinion before merging

**Review by ndelangen — APPROVED** [review]

**Pull request merged**
